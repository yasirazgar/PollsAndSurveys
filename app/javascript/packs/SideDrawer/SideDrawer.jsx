import React, { Component, Fragment } from 'react';

import Button from '../Utils/Button'
import ProfileForm from '../Profile/Form'
import SignInForm from '../Session/SignInForm'
import SignUpForm from '../Session/SignUpForm'
import Footer from '../Footer/Footer'

import './SideDrawer.scss';

import ProfileIcon from 'user__anonymous.svg'

// let isFormVisible = false;
// const showForm = () => {
//   let profileForm = document.getElementsByClassName("profile-form")[0];
//   profileForm.classList.add("show");
// }

class SideDrawer extends Component {
  state = {
    signedIn: true
  }

  showSignInForm = () => {
    let signInForm = document.getElementsByClassName("signin form")[0];
    let signUpForm = document.getElementsByClassName("signup form")[0];

    signUpForm.classList.add("hidden");
    signInForm.classList.remove("hidden");
  }

  showSignUpForm = () => {
    let signUpForm = document.getElementsByClassName("signup form")[0];
    let signInForm = document.getElementsByClassName("signin form")[0];

    signInForm.classList.add("hidden");
    signUpForm.classList.remove("hidden");
  }

  render () {
    let classes = 'side-drawer';
    if (this.props.open) {
      classes = classes + ' open';
    }

    let content;
    if (this.state.signedIn) {
      content = <Fragment>
        <Button classes="delete" text="Logout" />
        <div className="form">
          <img className="avatar" src={ProfileIcon}></img>
          <ProfileForm />
          <Button text="Submit"/>
        </div>
      </Fragment>
    }
    else {
      content = <Fragment>
        <Button key="0" classes="other" text="SignIn" clickHandler={this.showSignInForm} />,
        <Button key="1" classes="other" text="SignUp" clickHandler={this.showSignUpForm} />
        <div className="signin form">
          <SignInForm />
          <Button text="Login" />
        </div>
        <div className="signup form hidden">
          <SignUpForm />
          <Button text="Submit" />
        </div>
      </Fragment>
    }

    return (
      <nav className={classes}>
        <div className="wrapper">
          {content}

          <Footer />
        </div>
      </nav>
    );
  }
}

export default SideDrawer;
