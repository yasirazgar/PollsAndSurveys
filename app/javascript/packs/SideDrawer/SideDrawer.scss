.side-drawer {
  height: 100%;
  background: green;
  box-shadow: 1px 0px 7px rgba(0, 0, 0, 0.5);
  position: fixed;
  top: 0;
  left: 0;
  width: 70%;
  max-width: 400px;
  z-index: 200;
  transform: translateX(-100%);
  transition: transform 0.3s ease-out;
  &.open{
    transform: translateX(0);
  }

  .wrapper {
    height: 100%;
    list-style: none;
    display: flex;
    flex-direction: column;
    margin: 5px;
    .avatar {
      margin: 0.5rem 0;
      vertical-align: middle;
      width: 100px;
      height: 100px;
      border-radius: 50%;
      margin-left: auto;
      margin-right: auto;
      display: block;
    }
    .profile-form {
      input {
        width: 80%;
        padding: 5px 0px;
        margin: 5px 0;
        margin-left: auto;
        margin-right: auto;
        display: block;
      }
      // transform: translateX(-100%);
      // transition: transform 0.3s ease-out;
      // &.show{
      //   transform: translateX(0);
      // }
    }
    .btn {
      width: 80%;
      margin-left: auto;
      margin-right: auto;
      display: block;
    }
  }
}

@media (max-width: 700px) {
  .toolbar_navigation-items {
    display: none;
  }
}