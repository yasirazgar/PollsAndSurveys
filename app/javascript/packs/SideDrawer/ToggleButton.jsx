import React from 'react';
import PropTypes from 'prop-types';

import './ToggleButton.scss';

const ToggleButton = props => (
  <button className="toggle-button" onClick={props.clickHandler}>
    <div className="toggle-button__line" />
    <div className="toggle-button__line" />
    <div className="toggle-button__line" />
  </button>

);

ToggleButton.propTypes = {
  clickHandler: PropTypes.func
}

export default ToggleButton;
