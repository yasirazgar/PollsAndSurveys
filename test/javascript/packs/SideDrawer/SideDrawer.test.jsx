import React from 'react'
import { shallow } from 'enzyme'
import SideDrawer from 'packs/SideDrawer/SideDrawer'

describe("SideDrawer", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<SideDrawer />);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly with props, open: true", () => {
    const wrapper = shallow(<SideDrawer open={true} />)

    expect(wrapper).toMatchSnapshot();
  });
})
