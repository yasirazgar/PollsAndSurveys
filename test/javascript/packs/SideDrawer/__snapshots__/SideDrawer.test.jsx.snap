// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideDrawer renders correctly 1`] = `
<nav
  className="side-drawer"
>
  <ul>
    <li>
      <img
        className="avatar"
        src="test-file-stub"
      />
    </li>
    <li>
      <a
        href="/"
      >
        Logout/Login
      </a>
    </li>
  </ul>
</nav>
`;

exports[`SideDrawer renders correctly with props, open: true 1`] = `
<nav
  className="side-drawer open"
>
  <ul>
    <li>
      <img
        className="avatar"
        src="test-file-stub"
      />
    </li>
    <li>
      <a
        href="/"
      >
        Logout/Login
      </a>
    </li>
  </ul>
</nav>
`;
